packages/sdk-js/src/DidHelpers/createDid.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) 2018-2024, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import { Blockchain } from '@kiltprotocol/chain-helpers'
+import {
+  getFullDid,
+  getStoreTx,
+  type NewDidVerificationKey,
+  signingMethodTypes,
+} from '@kiltprotocol/did'
+import type { KiltAddress, SignerInterface } from '@kiltprotocol/types'
+import { Crypto, Signers } from '@kiltprotocol/utils'
+
+import { checkResultImpl, convertPublicKey, submitImpl } from './common.js'
+import type {
+  AcceptedPublicKeyEncodings,
+  SharedArguments,
+  TransactionHandlers,
+} from './interfaces.js'
+
+function implementsSignerInterface(input: any): input is SignerInterface {
+  return 'algorithm' in input && 'id' in input && 'sign' in input
+}
+
+/**
+ * Creates an on-chain DID based on an authentication key.
+ *
+ * @param options Any {@link SharedArguments} (minus `didDocument`) and additional parameters.
+ * @param options.fromPublicKey The public key that will feature as the DID's initial authentication method and will determine the DID identifier.
+ * @returns A set of {@link TransactionHandlers}.
+ */
+export function createDid(
```

Comment:
This is incorrect. This only adds a single key. We discussed adding ALL keys.

Reply:
No, in the design docs this was never meant to do more than simply create a DID based on an authentication key. All customisation is to be done after that.

Reply:
https://hackmd.io/@rflechtner/SkvnDDPVA

Reply:
adding batching would allow us to offer doing it all in one go though